drivers/rtc/rtc_ll_stm32.h

```diff
@@ -0,0 +1,318 @@
+
+	uint32_t ll_mask = 0;
+	/*
+	 * RTC Alarm LL mask should be set for all fields beyond the biggest one
+	 * that's being matched with RTC calendar to trigger alarm periodically
+	 */
+	if (id == RTC_STM32_ALRM_A) {
+		ll_mask = LL_RTC_ALMA_MASK_ALL;
+	}
+#if RTC_STM32_ALARMS_COUNT > 1
+	if (id == RTC_STM32_ALRM_B) {
+		ll_mask = LL_RTC_ALMB_MASK_ALL;
+	}
+#endif /* RTC_STM32_ALARMS_COUNT > 1 */
+
+	if (mask & RTC_ALARM_TIME_MASK_SECOND) {
+		if (id == RTC_STM32_ALRM_A) {
+			ll_mask &= ~LL_RTC_ALMA_MASK_SECONDS;
+		}
+#if RTC_STM32_ALARMS_COUNT > 1
+		if (id == RTC_STM32_ALRM_B) {
+			ll_mask &= ~LL_RTC_ALMB_MASK_SECONDS;
+		}
+#endif /* RTC_STM32_ALARMS_COUNT > 1 */
+
+		p_rtc_alrm_time->Seconds = bin2bcd(timeptr->tm_sec);
+	}
+
+	if (mask & RTC_ALARM_TIME_MASK_MINUTE) {
+		if (id == RTC_STM32_ALRM_A) {
+			ll_mask &= ~LL_RTC_ALMA_MASK_MINUTES;
+		}
+#if RTC_STM32_ALARMS_COUNT > 1
+		if (id == RTC_STM32_ALRM_B) {
+			ll_mask &= ~LL_RTC_ALMB_MASK_MINUTES;
+		}
+#endif /* RTC_STM32_ALARMS_COUNT > 1 */
+
+		p_rtc_alrm_time->Minutes = bin2bcd(timeptr->tm_min);
+	}
+
+	if (mask & RTC_ALARM_TIME_MASK_HOUR) {
+		if (id == RTC_STM32_ALRM_A) {
+			ll_mask &= ~LL_RTC_ALMA_MASK_HOURS;
+		}
+#if RTC_STM32_ALARMS_COUNT > 1
+		if (id == RTC_STM32_ALRM_B) {
+			ll_mask &= ~LL_RTC_ALMB_MASK_HOURS;
+		}
+#endif /* RTC_STM32_ALARMS_COUNT > 1 */
+
+		p_rtc_alrm_time->Hours = bin2bcd(timeptr->tm_hour);
+	}
+
+	if (mask & RTC_ALARM_TIME_MASK_WEEKDAY) {
+		/* the Alarm Mask field compares with the day of the week */
+		if (id == RTC_STM32_ALRM_A) {
+			ll_mask &= ~LL_RTC_ALMA_MASK_DATEWEEKDAY;
+			p_rtc_alarm->AlarmDateWeekDaySel = LL_RTC_ALMA_DATEWEEKDAYSEL_WEEKDAY;
+		}
+#if RTC_STM32_ALARMS_COUNT > 1
+		if (id == RTC_STM32_ALRM_B) {
+			ll_mask &= ~LL_RTC_ALMB_MASK_DATEWEEKDAY;
+			p_rtc_alarm->AlarmDateWeekDaySel = LL_RTC_ALMB_DATEWEEKDAYSEL_WEEKDAY;
+		}
+#endif /* RTC_STM32_ALARMS_COUNT > 1 */
+
+		if (timeptr->tm_wday == 0) {
+			/* sunday (tm_wday = 0) is not represented by the same value in hardware */
+			p_rtc_alarm->AlarmDateWeekDay = LL_RTC_WEEKDAY_SUNDAY;
+		} else {
+			/* all the other values are consistent with what is expected by hardware */
+			p_rtc_alarm->AlarmDateWeekDay = bin2bcd(timeptr->tm_wday);
+		}
+
+	} else if (mask & RTC_ALARM_TIME_MASK_MONTHDAY) {
+		/* the Alarm compares with the day number & ignores the day of the week */
+		if (id == RTC_STM32_ALRM_A) {
+			ll_mask &= ~LL_RTC_ALMA_MASK_DATEWEEKDAY;
+			p_rtc_alarm->AlarmDateWeekDaySel = LL_RTC_ALMA_DATEWEEKDAYSEL_DATE;
+		}
+#if RTC_STM32_ALARMS_COUNT > 1
+		if (id == RTC_STM32_ALRM_B) {
+			ll_mask &= ~LL_RTC_ALMB_MASK_DATEWEEKDAY;
+			p_rtc_alarm->AlarmDateWeekDaySel = LL_RTC_ALMB_DATEWEEKDAYSEL_DATE;
+		}
+#endif /* RTC_STM32_ALARMS_COUNT > 1 */
+
+		p_rtc_alarm->AlarmDateWeekDay = bin2bcd(timeptr->tm_mday);
+	}
+
+	p_rtc_alrm_time->TimeFormat = LL_RTC_TIME_FORMAT_AM_OR_24;
+
+	p_rtc_alarm->AlarmMask = ll_mask;
+
+	return 0;
+}
```

Comment:
This function is too long to be in a header.

Note that (unfortunately poorly named) ll_func functions used accross previous drivers are reserved to hw abstraction. 
My initial request to use these functions in previous patch set was targeting functions providing abstraction for F1 series. This is not the case anymore, so naming should be reviewed and functions can lie in driver c file when not providing abstraction.
This is not the case here, so you should use proper dedicated function names.